mesop/components/table/table.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from typing import Any, Protocol, runtime_checkable
+
+import mesop.components.table.table_pb2 as table_pb
+from mesop.component_helpers import insert_component, register_native_component
+
+
+@runtime_checkable
+class TableDataSource(Protocol):
+  """Interface for supporting different table data sources"""
+
+  def columns(self) -> list[str]:
+    """Returns a list of columns in the data source."""
+    pass
+
+  def to_data_source(self) -> list[table_pb.TableRow]:
+    """Returns the data source to match table_pb.TableType."""
+    pass
+
+
+@dataclass
+class TableDataSourcePandasDataFrame(TableDataSource):
+  # TODO: Need a Pydantic validator for pandas DataFrames. Temporarily using Any for now.
+  df: Any
+
+  def columns(self) -> list[str]:
+    return list(self.df.columns)
+
+  def to_data_source(self) -> list[table_pb.TableRow]:
+    columns = list(self.df.columns)
+    data = []
+    for df_row in self.df.itertuples():
+      row = {}
+      # TODO: Properly support the various Pandas data types.
+      for col_name in columns:
+        value = getattr(df_row, col_name)
+        if isinstance(value, bool):
+          row[col_name] = table_pb.TableValue(bool_value=value)
+        elif isinstance(value, str):
+          row[col_name] = table_pb.TableValue(string_value=value)
+        elif isinstance(value, int):
+          row[col_name] = table_pb.TableValue(int_value=value)
+        elif isinstance(value, float):
+          row[col_name] = table_pb.TableValue(float_value=value)
+        else:
+          raise ValueError(f"Unsupported Mesop table data type: {type(value)}")
+      data.append(table_pb.TableRow(row=row))
+    return data
+
+
+@register_native_component
```

Comment:
Not sure if validation is failing for you, but I had to skip validation for the plot component which also used a Protocol: https://github.com/google/mesop/blob/0d60e4a6627a508c9dc0e28b3631f553b9981ac3/mesop/components/plot/plot.py#L19

You may need to modify https://github.com/google/mesop/blob/0d60e4a6627a508c9dc0e28b3631f553b9981ac3/mesop/component_helpers/helper.py#L299 to allow accepting an argument for `skip_validation` a la https://github.com/google/mesop/blob/0d60e4a6627a508c9dc0e28b3631f553b9981ac3/mesop/component_helpers/helper.py#L100-L116

Reply:
I'm not really sure what's the right thing to do here. If we want to do a proper validation check (e.g. checking isinstance with the actual Pandas DataFrame class, or Matplotlib Figure), then we'll end up importing some pretty hefty third-party library into Mesop (which every Mesop developer will depend on, even if they don't use it).

I think skipping validation checking is reasonable - they'll probably get a more cryptic error, but in the future we could do something a little bit magical like check the class name via `__name__` (https://g.co/gemini/share/6ac3898f6693), but this feels a little bit too fancy for now.

Reply:
Cool. I'll just just skip the validation check. Definitely agree about the third party dependency issue.